### PR TITLE
Expose the underlying generator function from wrapper functions

### PR DESCRIPTION
**Pull request by ide**
Currently it's impossible to inspect the wrapped generator function if you want
to dynamically extract its source to produce documentation. With native async
functions this won't be a problem but right now the `co.wrap` wrapper is too
opaque for this use case.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,9 +24,11 @@ module.exports = co['default'] = co.co = co;
  */
 
 co.wrap = function (fn) {
-  return function () {
+  function createPromise() {
     return co.call(this, fn.apply(this, arguments));
-  };
+  }
+  createPromise.__generatorFunction__ = fn;
```
Comment (jonathanong) [inline index.js]: y `.__generatorFunction__` tho? y not just `.fn`?

Comment (ide) [inline index.js]: It's a semi-public API so I picked something more explicit and also less likely to have conflicts with libraries that add properties to Function.prototype.

**Comment by jonathanong** [comment]
anyone else care about the name?

**Comment by jonathanong** [comment]
f3003783cac8e496a124186164c9d7e21a57fa96

**Comment by jonathanong** [comment]
oops wrong PR :P

**Comment by jonathanong** [comment]
3004f22e31e0f4731f31fb8e0479756e2f97f210 thanks!

**Comment by ide** [comment]
Thanks :)

**Pull request closed without merging**